spring-vault-core/src/main/java/org/springframework/vault/authentication/AppRoleAuthentication.java

```diff
@@ -91,6 +96,48 @@ private VaultToken createTokenUsingAppRole() {
 		}
 	}
 
+	private String getRoleId() {
+		String roleId = options.getRoleId();
+		if (StringUtils.isEmpty(roleId) && !StringUtils.isEmpty(options.getAppRole())) {
```

Comment:
We should assume that `appRole` is set otherwise this case could return an empty RoleId.

Reply:
Does that mean I should remove the appRole verification there and rely on the Assertions from AppRoleAuthenticationOptions ?

Reply:
Yes. We expect a valid options object when constructing an `AppRoleAuthentication` instance.